crowbar_framework/app/models/updater_service.rb

```diff
@@ -54,16 +62,22 @@ def create_proposal
   end
 
   def apply_role_pre_chef_call(old_role, role, all_nodes)
-    @logger.debug("Updater apply_role_post_chef_call: entering #{all_nodes.inspect}")
-    # Remove [:updater][:one_shot_run] flag from node
+    @logger.debug("Updater apply_role_pre_chef_call: entering #{all_nodes.inspect}")
+    # Remove [:updater][...] flags from node
     all_nodes.each do |n|
       node = NodeObject.find_node_by_name n
       unless node[:updater].nil?
-        node[:updater][:one_shot_run] = false
-        @logger.debug("Updater apply_role_post_chef_call: delete [:updater][:one_shot_run] for #{node.name} (#{node[:updater].inspect}")
-        node.save
+        if node.role? "updater"
+          node[:updater][:one_shot_run] = false
+          @logger.debug("Updater apply_role_post_chef_call: delete [:updater][:one_shot_run] for #{node.name} (#{node[:updater].inspect}")
+          node.save
+        end
+        if node.role? "updater-crowbar-upgrade"
```

Comment:
I think this part needs to live outside the ```unless node[:updater].nil?``` branch. AFAICS ```node[:updater]``` is only there after the updater barclamp was applied at least once. And the other code in that ```unless``` clause is only needed for cleaning up the ```one_shot``` stuff